chia/rpc/wallet_rpc_api.py

```diff
@@ -624,13 +625,15 @@ async def create_new_wallet(self, request: Dict) -> EndpointResult:
                 did_id: Optional[bytes32] = None
                 if "did_id" in request and request["did_id"] is not None:
                     did_id = decode_puzzle_hash(request["did_id"])
-                if wallet.type() == WalletType.NFT and wallet.get_did() == did_id:
-                    log.info("NFT wallet already existed, skipping.")
-                    return {
-                        "success": True,
-                        "type": wallet.type(),
-                        "wallet_id": wallet.id(),
-                    }
+                if wallet.type() == WalletType.NFT:
+                    assert isinstance(wallet, NFTWallet)
```

Comment:
Why not just do the type check as the if?

Also, I don't mean to request this change here, but maybe this behavior can be shifted onto the wallets?

Reply:
I agree with both. We should use the language mechanism to check types, to let mypy (and other generic python programmers) keep up.

I consider this patch a first step to highlight this need.